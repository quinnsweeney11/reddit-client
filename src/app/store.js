import { configureStore } from '@reduxjs/toolkit';
import counterReducer from '../features/counter/counterSlice';
import postContainerReducer from '../containers/postContainer/postContainerSlice';

export const store = configureStore({
  reducer: {
    counter: counterReducer,
    postContainer: postContainerReducer
  },
});
