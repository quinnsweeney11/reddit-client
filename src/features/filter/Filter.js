import React, {useEffect} from 'react';
import { useSelector, useDispatch } from 'react-redux';
import './filter.css';
import { switchFilter } from './filterSlice';
import { getRedditData } from '../../containers/postContainer/postContainerSlice';


export function Filter(props) {
    const dispatch = useDispatch();
    const { currentFilter } = useSelector((state) => state.filter)
    console.log(currentFilter)
    useEffect(() => {
        const url = `https://www.reddit.com/${currentFilter}.json`
        dispatch(getRedditData(url))
    })
    return(
        <div className="filter">
            <div className='buttons'>
                <button className={currentFilter === 'hot' ? 'active' : 'inactive'} onClick={() => dispatch(switchFilter({filter:'hot'}))}>Hot</button>
                <button className={currentFilter === 'new' ? 'active' : 'inactive'} onClick={() => dispatch(switchFilter({filter: 'new'}))}>New</button>
                <button className={currentFilter === 'best' ? 'active' : 'inactive'} onClick={() => dispatch(switchFilter({filter: 'best'}))}>Best</button>
            </div>
            <div className='text'>
                <h3>This project is still a work in progress, here are some of the things to be added in the future:</h3>
                <ul>
                    <li>Loading more posts when a user's scroll reaches the bottom of the page</li>
                    <li>Search bar to filter by subreddit</li>
                    <li>Interface to browse comments on posts</li>
                    <li>Improved mobile experience</li>
                </ul>
                <p class="myNote">Made with ❤️ by Quinn Sweeney</p>
            </div>
        </div>
    )
}