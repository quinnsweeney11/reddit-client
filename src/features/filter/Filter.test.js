import React from "react";
import { Filter } from "./Filter";
import renderer from "react-test-renderer";
import { Provider } from "react-redux";
import { store } from "../../app/store";

describe("snapshot", () => {
  it("main filter", () => {
    const component = renderer.create(
      <Provider store={store}>
        <Filter />
      </Provider>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
