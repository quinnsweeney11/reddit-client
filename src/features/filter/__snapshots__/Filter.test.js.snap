// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot main filter 1`] = `
<div
  className="filter"
>
  <div
    className="buttons"
  >
    <button
      className="active"
      onClick={[Function]}
    >
      Hot
    </button>
    <button
      className="inactive"
      onClick={[Function]}
    >
      New
    </button>
    <button
      className="inactive"
      onClick={[Function]}
    >
      Best
    </button>
  </div>
  <div
    className="text"
  >
    <h3>
      This project is still a work in progress, here are some of the things to be added in the future:
    </h3>
    <ul>
      <li>
        Loading more posts when a user's scroll reaches the bottom of the page
      </li>
      <li>
        Search bar to filter by subreddit
      </li>
      <li>
        Interface to browse comments on posts
      </li>
      <li>
        Improved mobile experience
      </li>
    </ul>
    <p
      class="myNote"
    >
      Made with ❤️ by Quinn Sweeney
    </p>
  </div>
</div>
`;
